core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -83,11 +85,31 @@ class CloudSqlInstance {
       CredentialFactory tokenSourceFactory,
       ListeningScheduledExecutorService executor,
       ListenableFuture<KeyPair> keyPair) {
+    this(
+        connectionName,
+        instanceDataSupplier,
+        authType,
+        tokenSourceFactory,
+        executor,
+        keyPair,
+        RateLimiter.burstyBuilder(2, Duration.ofSeconds(30)).build());
+  }
+
+  /** A constructor for use in tests that allows the caller to set the RateLimiter. */
+  CloudSqlInstance(
```

Comment:
Let's just update the constructor to pass in the rate limiter. We don't need two constructors.

Reply:
As a rule, I think we shouldn't be making test only APIs. And parameterizing the rate limiter is actually useful. Imagine if we wanted to expose rate limiting knobs to the user, or configure instances with different rate limits.

Reply:
Fixed. I'm not convinced, but it's not worth arguing about.